java/ql/src/Security/CWE/CWE-918/RequestForgery.qll

```diff
@@ -5,7 +5,12 @@ import semmle.code.java.frameworks.spring.Spring
 import semmle.code.java.frameworks.JaxWS
```

Comment:
We aim to support extensibility in the security queries, by putting a `.qll` in `semmle.code.java.Security` that includes a couple of abstract classes for the most common extensions - in this case I think we should expose sinks, additional steps, and sanitizers for customization.

To do this, I think this entire file should be split in two parts. The ad-hoc and very query specific things can move to the `.ql` whereas the parts that are needed to support this kind of extension should move to the importable dir.

Reply:
In particular, the `abstract class RequestForgerySink` and `abstract class RequestForgerySanitizer` should be importable. (And their private subclasses should likely live in the file next to those definitions).